macros/activity_schema/activity/generate_activity_yml.sql

```diff
@@ -0,0 +1,74 @@
+-- Get data_types object from model config block
+{% macro get_activity_config(model) -%}
+    {% set relation = ref(model) %}
+    {% for node in graph.nodes.values()
+        | selectattr("resource_type", "equalto", "model")
+        | selectattr("name", "equalto", relation.identifier) %}
+        {% do return(node.config) %}
+    {% endfor %}
+{%- endmacro %}
+
+{% macro generate_activity_yml(activities) %}
+
+{% set yaml_output = [] %}
+
+{% do yaml_output.append('version: 2') %}
+{% do yaml_output.append('models:') %}
+
+-- Loop through each activity
+{% for activity in activities %}
+
+  {% do yaml_output.append('  - name: ' ~ activity) %}
+  {% do yaml_output.append('    columns:') %}
+
+  {% set columns = [
+      {'name': 'activity_id', 'description': 'Unique identifier for the activity.', 'data_type': 'STRING', 'tests': ['unique', 'not_null']},
+      {'name': 'customer_id', 'description': 'Identifier for the customer.', 'data_type': 'STRING'},
```

Comment:
Do you think it's worth changing the description to be `identifier for the entity` to support cases where the entity is _not_ a generic "customer"?